File: app/src/main/java/echo/com/surveys/adapter/SurveyFragmentPagerAdapter.java
package echo.com.surveys.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import echo.com.surveys.fragment.SurveyFragment;
import echo.com.surveys.model.SurveyModel;

import java.util.List;

public class SurveyFragmentPagerAdapter extends FragmentStatePagerAdapter {
    private List<SurveyModel> surveys;

    public SurveyFragmentPagerAdapter(FragmentManager fm, List<SurveyModel> surveys) {
        super(fm);
        this.surveys = surveys;
    }

    @Override
    public Fragment getItem(int position) {
        return SurveyFragment.getInstance(surveys.get(position));
    }

    @Override
    public int getCount() {
        return surveys.size();
    }
}

File: app/src/main/java/echo/com/surveys/adapter/SurveyFragmentPagerAdapter.kt
package echo.com.surveys.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import echo.com.surveys.fragment.SurveyFragment
import echo.com.surveys.model.SurveyModel

class SurveyFragmentPagerAdapter(fm: androidx.fragment.app.FragmentManager, private val surveys: MutableList<SurveyModel>) :
    androidx.fragment.app.FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): androidx.fragment.app.Fragment {
        return SurveyFragment.getInstance(surveys[position])
    }

    override fun getCount(): Int {
        return surveys.size
    }

    fun addItems(newItems: List<SurveyModel>?) {
        if (newItems != null && newItems.isNotEmpty()) {
            surveys.clear()
            surveys.addAll(newItems)
            notifyDataSetChanged()
        }
    }
}

